src/main/java/racing/domain/Cars.java

```diff
@@ -32,4 +30,36 @@ public Cars playOnceTurn() {
         }
         return new Cars(movedCars);
     }
+
+    public List<String> findWinners() {
+
+        List<String> winners = new ArrayList<>();
+        int maxLocation = maxLocation();
+
+        for (int i = 0; i < this.cars.size(); i++) {
+            addOnlyWinner(winners, maxLocation, i);
+        }
+        return winners;
+    }
+
+    private void addOnlyWinner(List<String> winners, int maxLocation, int i) {
+        if (checkWinner(maxLocation, this.cars.get(i))) {
+            winners.add(this.cars.get(i).name());
+        }
+    }
+
+    private boolean checkWinner(int maxLocation, Car car) {
+        if (car.isWinner(maxLocation)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
```suggestion
    public List<String> findWinners() {
        int maxLocation = maxLocation();
        return cars.stream()
                .filter(car -> car.isWinner(maxLocation))
                .map(Car::name)
                .collect(Collectors.toList());
    }
```
- 질문주신 로직은 위와 같이 Stream API를 활용하여 리팩토링할 수 있지 않을까요